src/matrix_cookbook/mat_eq27_lib.lean

```diff
@@ -0,0 +1,235 @@
+/-
+Copyright (c) 2023 Mohanad Ahmed, Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed, Eric Wieser
+-/
+import linear_algebra.matrix.trace
+import linear_algebra.matrix.determinant
+import data.matrix.notation
+import data.fintype.big_operators
+import tactic.ring
+import tactic.norm_fin
+import data.is_R_or_C.basic
+import algebra.char_p.basic
+
+/-!
+#  Traces and Determinants of 1st, 2nd and 3rd Powers of 4x4 Matrices
+
+This file contains lemmas in rather verbose form of matrix fin 4 fin 4 R.
+
+These are used to prove equation 27 in the matrix cookbook. 
+
+The results are all field results. The last one `eq_27_before_last` which
+involves multiplication and division by the numbers 2, 3 and 6 imposes more
+requirement, which I satsified by requiring a field with characteristic zero.
+
+-/
+
+open_locale matrix big_operators
+open matrix
+
+variables {R : Type*}[field R][char_zero R]
+
+lemma trace_fin_four {A : matrix (fin 4) (fin 4) R} :
```

Comment:
Can you move this lemma and `det_fin_four` to `for_mathlib/data/matrix.lean`?

Reply:
(the other lemmas are rather specialized, but I think these do belong in mathlib)